bazel/testing/lit_autoupdate_base.py

```diff
@@ -0,0 +1,408 @@
+    check_lines = [
+        CheckLine(out_line, line_number_format, line_number_pattern)
+        for out_line in out_lines
+    ]
+
+    result_lines: List[Line] = []
+    # CHECK lines must go after AUTOUPDATE.
+    while orig_lines and orig_lines[0].line_number <= autoupdate_line_number:
+        result_lines.append(orig_lines.pop(0))
+    # Interleave the original lines and the CHECK: lines.
+    while orig_lines and check_lines:
+        # Original lines go first when the CHECK line is known and later.
+        if (
+            check_lines[0].line_numbers
+            and check_lines[0].line_numbers[0] > orig_lines[0].line_number
+        ):
+            result_lines.append(orig_lines.pop(0))
+        else:
+            check_line = check_lines.pop(0)
+            # Indent to match the next original line.
+            check_line.indent = re.findall("^ *", orig_lines[0].text)[0]
+            result_lines.append(check_line)
+    # One list is non-empty; append remaining lines from both to catch it.
+    result_lines.extend(orig_lines)
+    result_lines.extend(check_lines)
+
+    return result_lines
+
+
+def update_check(update_args: UpdateArgs, test: str) -> bool:
+    """Updates the CHECK: lines for `test` by running explorer.
+
+    Returns true if a change was made.
+    """
+    with open(test) as f:
+        orig_lines = f.readlines()
+
+    # Make sure we're supposed to autoupdate.
+    autoupdate = find_autoupdate(test, orig_lines)
+    if autoupdate is None:
+        return False
+
+    # Determine the merged output lines.
+    out_lines = get_actual_output(
+        update_args, test, autoupdate.cmd, update_args.extra_check_replacements
+    )
+    result_lines = merge_lines(
+        update_args.line_number_format,
+        update_args.line_number_pattern,
+        autoupdate.line_number,
+        orig_lines,
+        out_lines,
+    )
+
+    # Calculate the remap for original lines.
+    line_number_remap = dict(
+        [
+            (line.line_number, i)
+            for i, line in enumerate(result_lines)
+            if isinstance(line, OriginalLine)
+        ]
+    )
+    # If the last line of the original output was a CHECK, replace it with an
+    # empty line.
+    if orig_lines[-1].lstrip().startswith("// CHECK"):
+        line_number_remap[len(orig_lines) - 1] = len(result_lines) - 1
+
+    # Generate contents for any lines that depend on line numbers.
+    formatted_result_lines = [
+        line.format(output_line_number=i, line_number_remap=line_number_remap)
+        for i, line in enumerate(result_lines)
+    ]
+
+    # If nothing's changed, we're done.
+    if formatted_result_lines == orig_lines:
+        return False
+
+    # Interleave the new CHECK: lines with the tested content.
+    with open(test, "w") as f:
+        f.writelines(formatted_result_lines)
+        return True
+
+
+def update_checks(update_args: UpdateArgs, tests: Set[str]) -> None:
+    """Runs bazel to update CHECK: lines in lit tests."""
```

Comment:
```suggestion
    """Updates CHECK: lines in lit tests."""
```
I don't think this runs `bazel`.